tensorflow_addons/losses/giou_loss_test.py

```diff
@@ -0,0 +1,84 @@
+"""Tests for GIoU loss."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from absl.testing import parameterized
+
+import numpy as np
+import tensorflow as tf
+from tensorflow_addons.utils import test_utils
+from tensorflow_addons.losses import giou_loss, GIoULoss
+
+
+@test_utils.run_all_in_graph_and_eager_modes
+class GIoULossTest(tf.test.TestCase, parameterized.TestCase):
+    """GIoU test class."""
+
+    def test_config(self):
+        gl_obj = GIoULoss(
+            reduction=tf.keras.losses.Reduction.NONE, name='giou_loss')
+        self.assertEqual(gl_obj.name, 'giou_loss')
+        self.assertEqual(gl_obj.reduction, tf.keras.losses.Reduction.NONE)
+
+    @parameterized.named_parameters(("float16", np.float16),
+                                    ("float32", np.float32),
+                                    ("float64", np.float64))
+    def test_iou(self, dtype):
+        boxes1 = tf.constant([[4.0, 3.0, 7.0, 5.0], [5.0, 6.0, 10.0, 7.0]],
+                             dtype=dtype)
+        boxes2 = tf.constant([[3.0, 4.0, 6.0, 8.0], [14.0, 14.0, 15.0, 15.0]],
+                             dtype=dtype)
+        expected_result = tf.constant([14.0 / 16.0, 1.], dtype=dtype)
+        loss = giou_loss(boxes1, boxes2, mode='iou')
+        self.assertAllCloseAccordingToType(loss, expected_result)
+
+    @parameterized.named_parameters(("float16", np.float16),
+                                    ("float32", np.float32),
+                                    ("float64", np.float64))
+    def test_giou_loss(self, dtype):
+        boxes1 = tf.constant([[4.0, 3.0, 7.0, 5.0], [5.0, 6.0, 10.0, 7.0]],
+                             dtype=dtype)
+        boxes2 = tf.constant([[3.0, 4.0, 6.0, 8.0], [14.0, 14.0, 15.0, 15.0]],
+                             dtype=dtype)
+        expected_result = tf.constant(
+            [1.07500000298023224, 1.9333333373069763], dtype=dtype)
+        loss = giou_loss(boxes1, boxes2)
+        self.assertAllCloseAccordingToType(loss, expected_result)
+
+    @parameterized.named_parameters(("float16", np.float16),
+                                    ("float32", np.float32),
+                                    ("float64", np.float64))
+    def test_giou_loss_with_dims(self, dtype):
+        boxes1 = tf.constant([[[4.0, 3.0, 7.0, 5.0], [5.0, 6.0, 10.0, 7.0]],
+                              [[4.0, 3.0, 7.0, 5.0], [5.0, 6.0, 10.0, 7.0]]],
+                             dtype=dtype)
+        boxes2 = tf.constant(
+            [[[3.0, 4.0, 6.0, 8.0], [14.0, 14.0, 15.0, 15.0]],
+             [[3.0, 4.0, 6.0, 8.0], [14.0, 14.0, 15.0, 15.0]]],
+            dtype=dtype)
+        expected_result = tf.constant(
+            [[1.07500000298023224, 1.9333333373069763],
+             [1.07500000298023224, 1.9333333373069763]],
+            dtype=dtype)
+        loss = giou_loss(boxes1, boxes2)
+        self.assertAllCloseAccordingToType(loss, expected_result)
+
```

Comment:
Could you also add a test to check if the loss is compatible with `tf.keras.Model.compile()`? Like what's the shape of input/output you use as example in the docstring.